css/src/components/button.scss

```diff
@@ -0,0 +1,324 @@
+@use 'sass:map';
+
+$button-color: $text !default;
+$button-background-color: $body-background !default;
+
+$button-large-font-size: $font-size-6 !default;
+$button-small-font-size: $font-size-8 !default;
+$button-x-small-font-size: $font-size-9 !default;
+
+$button-border-color: $text-subtle !default;
+$button-border-width: $control-border-width !default;
+$button-border-radius: $radius-small;
+
+$button-padding-vertical: calc(0.375em - #{$button-border-width}) !default;
+$button-padding-horizontal: 0.75em !default;
+
+$button-extra-small-padding-vertical: calc(0.375em - #{$button-border-width}) !default;
+$button-extra-small-padding-horizontal: 1em !default;
+
+$button-hover-color: $text !default;
+$button-hover-border-color: $default-hover !default;
+$button-hover-background-color: $default-hover-invert !default;
+
+$button-focus-color: $text !default;
+$button-focus-border-color: $primary !default;
+
+$button-active-color: $text !default;
+$button-active-border-color: $text !default;
+
+$button-text-color: $text !default;
+$button-text-hover-background-color: $body-background-medium !default;
+$button-text-hover-color: $text !default;
+$button-text-active-background-color: $body-background-medium !default;
+
+$button-disabled-background-color: $body-background-medium !default;
+$button-disabled-shadow: none !default;
+$button-disabled-opacity: 0.5 !default;
+
+@mixin button-x-small {
+	padding: $button-extra-small-padding-vertical $button-extra-small-padding-horizontal;
+	border-radius: $radius-small;
+	font-size: $button-x-small-font-size;
+
+	.icon {
+		height: $button-x-small-font-size;
```

Comment:
Typically icons use em sizing so this would not be required.

Reply:
You're speaking of `height`? I think it does required it because the major point of icon is to ensure the space is there _before__ external resources are rendered. No?